File: typescript/word-count/word-count.ts

class Words {
    count(phrase: string): Map<string, number> {
        let m = new Map();
        let words = phrase.trim().split(/\s+/g)

        words.forEach(function (w) {
            w = w.toLowerCase()
            if (m.has(w)) {
                m.set(w, m.get(w) + 1);
            } else {
                m.set(w, 1);
            }
        })
        return m;
    }
}

export default Words

File: go/word-count/word_count.go
// Package wordcount provides function to calculate count
// of words
package wordcount

import (
	"strings"
	"unicode"
)

type Frequency map[string]int

// WordCount returns Frequency map with counts of words from input
func WordCount(input string) Frequency {
	splitFn := func(c rune) bool {
		return !unicode.IsNumber(c) && !unicode.IsLetter(c) && c != '\''
	}

	wordCandidates := strings.FieldsFunc(input, splitFn)

	res := make(Frequency)

	for _, candidate := range wordCandidates {
		word := strings.Trim(strings.ToLower(candidate), "'")
		res[word]++
	}

	return res
}
